sql/moz-fx-data-shared-prod/fenix_derived/firefox_android_clients_v2/query.sql

```diff
@@ -0,0 +1,224 @@
+    AND client_info.client_id IS NOT NULL
+),
+first_session_ping AS (
+  SELECT
+    client_id,
+    sample_id,
+    ARRAY_AGG(
+      IF(
+        adjust_ad_group IS NOT NULL
+        OR adjust_campaign IS NOT NULL
+        OR adjust_creative IS NOT NULL
+        OR adjust_network IS NOT NULL,
+        STRUCT(
+          submission_timestamp,
+          adjust_ad_group,
+          adjust_campaign,
+          adjust_creative,
+          adjust_network
+        ),
+        NULL
+      ) IGNORE NULLS
+      ORDER BY
+        submission_timestamp ASC
+      LIMIT
+        1
+    )[SAFE_OFFSET(0)] AS adjust_info,
+  FROM
+    first_session_ping_base
+  GROUP BY
+    client_id,
+    sample_id
+),
+-- Find earliest data per client from the metrics ping.
+metrics_ping_base AS (
+  SELECT
+    client_info.client_id AS client_id,
+    sample_id,
+    submission_timestamp,
+    NULLIF(fenix_metrics.metrics.string.metrics_adjust_ad_group, "") AS adjust_ad_group,
+    NULLIF(fenix_metrics.metrics.string.metrics_adjust_campaign, "") AS adjust_campaign,
+    NULLIF(fenix_metrics.metrics.string.metrics_adjust_creative, "") AS adjust_creative,
+    NULLIF(fenix_metrics.metrics.string.metrics_adjust_network, "") AS adjust_network,
+  FROM
+    fenix.metrics AS fenix_metrics
+  WHERE
+    DATE(submission_timestamp) = @submission_date
+    AND client_info.client_id IS NOT NULL
+),
+metrics_ping AS (
+  SELECT
+    client_id,
+    sample_id,
+    ARRAY_AGG(
+      IF(
+        adjust_ad_group IS NOT NULL
+        OR adjust_campaign IS NOT NULL
+        OR adjust_creative IS NOT NULL
+        OR adjust_network IS NOT NULL,
+        STRUCT(
+          submission_timestamp,
+          adjust_ad_group,
+          adjust_campaign,
+          adjust_creative,
+          adjust_network
+        ),
+        NULL
+      ) IGNORE NULLS
+      ORDER BY
+        submission_timestamp ASC
+      LIMIT
+        1
+    )[SAFE_OFFSET(0)] AS adjust_info,
+  FROM
+    metrics_ping_base
+  GROUP BY
+    client_id,
+    sample_id
+),
+_current AS (
+  SELECT
+    client_id,
+    sample_id,
+    first_seen_date,
+    first_reported_country,
+    first_reported_isp,
+    channel,
+    device_manufacturer,
+    device_model,
+    os_version,
+    app_version,
+    COALESCE(first_session.adjust_info, metrics.adjust_info) AS adjust_info,
+    STRUCT(
+      IF(first_session.client_id IS NULL, FALSE, TRUE) AS is_reported_first_session_ping,
+      IF(metrics.client_id IS NULL, FALSE, TRUE) AS is_reported_metrics_ping,
+      CASE
+        WHEN first_session.adjust_info IS NOT NULL
+          THEN "first_session"
+        WHEN metrics.adjust_info IS NOT NULL
+          THEN "metrics"
+        ELSE NULL
+      END AS adjust_info__source_ping
+    ) AS metadata,
+    -- field to help us identify suspicious devices on iOS, for more info see: bug-1846554
+    (app_version = '107.2' AND submission_date >= '2023-02-01') AS is_suspicious_device_client,
```

Comment:
this shouldn't be here, the incident is iOS only

Reply:
oh yes, fair point. I thought I deleted this.